### ISSUE TITLE
Oracle table having column datatype as a user defined type object

### ISSUE THREAD

**Issue by friendarora** [laurenz/oracle_fdw#249]
Hello,

i am able to connect to a oracle database from postgres using the Oracle_fdw
but my oracle table is having column with datatype as a user defined TYPE object
i am neither able to select this column nor i am able to insert any record in this table

I there a way in oracle_fdw to access this column value and insert a record into the foreign table

**Comment by laurenz** [comment]
Yes, oracle_fdw cannot handle user defined data types.

Let's assume the Oracle table looks like that:

    CREATE TABLE badcol (
       id number(5) PRIMARY KEY,
       bad my_custom_type,
       val VARCHAR2(20)
    );

There are several ways:

1. You don't need the `val` column (or anything after the user defined column:
   Then you can

       CREATE FOREIGN TABLE badcol (id integer NOT NULL)
          SERVER oracle OPTIONS (table 'BADCOL');

   Inserts will insert `bad` and `val` as NULL.

2. Create a view on the Oracle side that omits the column or casts it to `VARCHAR2` and define the foreign table on that. If you omit the column, DML will work, otherwise not.

3. Define the foreign table like this:

       CREATE FOREIGN TABLE badcol (id integer NOT NULL, val varchar(20))
          SERVER oracle OPTIONS (table '(SELECT id, val FROM badcol)');

   Then DML will work and leave `bad` NULL.
   Of course you can also convert `bad` to `VARCHAR2` in that query, then you can select it, but not modify the table.

**Comment by friendarora** [comment]
hello,

Since our requirement was to insert into the custom field of the foreign table using Oracle FDW we could not take any of the options mentioned above.

Though we found a different workaround to our problem
1. we created a new table in oracle database with no custom TYPE field but included all the individual fields required in the custom TYPE 
2. Created a DML trigger on this new table to insert the data into our original oracle table having custom TYPE field
2. imported the new table into postgres as foreign table rather than the table having TYPE field through oracle fdw
3. inserted into this new foreign table from postgres via oracle_fdw
4. because of the trigger the data got inserted into the main oracle table
5. As trigger was firing from one oracle table to another oracle table it was able to access custom TYPE
